Consumer/MessageConsumer.php

```diff
@@ -45,10 +61,10 @@ public function __construct(MessageManager $messageManager)
      */
     public function execute(AMQPMessage $msg)
     {
-        $rawMessage = json_decode($msg->body);
-        $message = $this->messageManager->createMessageFromData($rawMessage);
-        $this->messageManager->dispatchMessage($message);
+        /** @var Message $message */
+        $message = $this->serializer
+            ->deserialize($msg->body, 'Ecentria\Libraries\EcentriaAPIEventsBundle\Model\Message', 'json');
```

Comment:
1) The Model\Message class should be configurable with 'Ecentria\Libraries\EcentriaAPIEventsBundle\Model\Message' as the default.
2) We should create an Interface that implements the "getSource" method since it is the only method we absolutely require
3) we should require that the configured class implements the above interface
4) does this deserialize throw an exception if the provided data does not fit at all with the provided data?